src/battle.c

```diff
@@ -0,0 +1,446 @@
+    asm("adds r1, r7, r3");
+    asm("strb r0, [r1]");
+    asm("strb r4, [r1, #6]");
+    asm("adds r0, #0x5e");
+    asm("adds r1, r7, r0");
+    asm("movs r0, #0xff");
+    asm("strb r0, [r1]");
+    asm("strb r4, [r1, #6]");
+    asm("adds r3, #2");
+    asm("adds r1, r7, r3");
+    asm("strb r0, [r1]");
+    asm("strb r4, [r1, #6]");
+    asm("adds r0, #0x60");
+    asm("adds r1, r7, r0");
+    asm("movs r0, #0xff");
+    asm("strb r0, [r1]");
+    asm("strb r4, [r1, #6]");
+    asm("movs r0, #6");
+    asm("strb r0, [r2]");
+    asm("movs r1, #0xbd");
+    asm("lsls r1, r1, #1");
+    asm("adds r0, r7, r1");
+    asm("strb r4, [r0]");
+    asm("adds r3, #0x1d");
+    asm("adds r0, r7, r3");
+    asm("strb r4, [r0]");
+    asm("adds r1, #2");
+    asm("adds r0, r7, r1");
+    asm("strb r4, [r0]");
+    asm("movs r3, #0xdc");
+    asm("lsls r3, r3, #2");
+    asm("adds r0, r7, r3");
+    asm("str r6, [r0]");
+    asm("str r6, [sp]");
+    asm("movs r0, #0x88");
+    asm("lsls r0, r0, #2");
+    asm("adds r1, r7, r0");
+    asm("ldr r2, _0803E710 @ =0x0500003A");
+    asm("mov r0, sp");
+    asm("bl CpuSet");
+    asm("movs r1, #0xc2");
+    asm("lsls r1, r1, #2");
+    asm("adds r0, r7, r1");
+    asm("strh r6, [r0]");
+    asm("ldr r3, _0803E714 @ =0x0000030A");
+    asm("adds r0, r7, r3");
+    asm("strh r6, [r0]");
+    asm("adds r1, #4");
+    asm("adds r0, r7, r1");
+    asm("bl sub_08009F0C");
+    asm("movs r3, #0xcd");
+    asm("lsls r3, r3, #2");
+    asm("adds r5, r7, r3");
+    asm("adds r0, r5, #0");
+    asm("bl sub_08009F0C");
+    asm("adds r0, r5, #0");
+    asm("movs r1, #0");
+    asm("movs r2, #1");
+    asm("bl sub_0800A6C0");
+    asm("adds r0, r5, #0");
+    asm("movs r1, #2");
+    asm("movs r2, #2");
+    asm("bl sub_0800A6C4");
+    asm("adds r0, r5, #0");
+    asm("movs r1, #1");
+    asm("bl sub_0800A728");
+    asm("mov r0, r8");
+    asm("bl sub_08009F0C");
+    asm("mov r0, r8");
+    asm("movs r1, #6");
+    asm("movs r2, #2");
+    asm("bl sub_0800A6C4");
+    asm("adds r0, r7, #0");
+    asm("bl sub_08049714");
+    asm("movs r1, #0xb3");
+    asm("lsls r1, r1, #1");
+    asm("adds r0, r7, r1");
+    asm("strb r4, [r0]");
+    asm("movs r3, #0xe3");
+    asm("lsls r3, r3, #2");
+    asm("adds r0, r7, r3");
+    asm("strb r4, [r0]");
+    asm("adds r1, #0x1a");
+    asm("adds r0, r7, r1");
+    asm("str r6, [r0]");
+    asm("add sp, #4");
+    asm("pop {r3}");
+    asm("mov r8, r3");
+    asm("pop {r4, r5, r6, r7}");
+    asm("pop {r0}");
+    asm("bx r0");
+    asm(".align 2, 0");
+    asm("_0803E70C: .4byte 0x0000038B");
+    asm("_0803E710: .4byte 0x0500003A");
+    asm("_0803E714: .4byte 0x0000030A");
+    asm(".syntax divided");
+}
+
+#endif
+
+/*
+I don't think this can be matched under current circumstance. 
+we need to figure out the data structure of BattleAttrib.
+https://cexplore.karathan.at/z/XMFsqu
+I pull this mainly because some fields of BattleAttrib can be documented.
+*/
```

Comment:
I believe it's the time to match or at least improve this so that we won't get into trouble later. A little work on it: https://decomp.me/scratch/YpJBg